src/d3d12/d3d12_acceleration_structure..cpp

```diff
@@ -343,16 +385,32 @@ namespace wr::d3d12
 	void DestroyAccelerationStructure(AccelerationStructure& structure)
 	{
 		SAFE_RELEASE(structure.m_scratch);
-		SAFE_RELEASE(structure.m_native);
-		SAFE_RELEASE(structure.m_instance_desc);
+
+		for (auto& as : structure.m_natives)
+		{
+			SAFE_RELEASE(as);
+		}
+		for (auto& inst_desc : structure.m_instance_descs)
+		{
+			SAFE_RELEASE(inst_desc);
+		}
+	}
+
+	void UAVBarrierAS(CommandList* cmd_list, AccelerationStructure structure, std::uint32_t frame_idx)
```

Comment:
can this be a reference to AccelerationStructure?

Reply:
No thanks to fallback, but can be const ref or I need to cast away const.